src/core/main.c

```diff
@@ -294,6 +295,27 @@ static int parse_crash_chvt(const char *value) {
         return 0;
 }
 
+static int parse_confirm_spawn(const char *value, char **console) {
+        int r;
+
+        if (value) {
+                r = parse_boolean(value);
+                if (r == 0) {
+                        *console = NULL;
+                        return 0;
+                }
+                if (r > 0)
+                        value = "/dev/console";
+        } else
+                value = "/dev/console";
+
+        if (!is_path(value))
+                value = strjoina("/dev/", value);
+
+        *console = strdup(value);
```

Comment:
as a matter of coding style, please do not clobber return values on failure... i.e. write the result of strdup() into some temporary var first, and write it to *console only on success. See CODING_STYLE about this. 

Also I think it would be nice to suppress one allocation here:

```c
if (value) 
        r = parse_boolean(value);
else
        r = 1;

if (r == 0) { /* off */
        *console = NULL;
        return 0;
} else if (r > 0) /* on with default tty */
        s = strdup("/dev/console");
else if (is_path(value)) /* on with fully qualified path */
        s = strdup(value);
else /* on with only a tty file name, not a fully qualified path */
        s = strjoin("/dev/", value, NULL);
if (!s)
        return -ENOMEM;
*console = s;
return 0;
```